src/Pulsar.Client/Auth/Oauth2/AuthenticationOauth2.fs

```diff
@@ -81,6 +80,32 @@ type internal AuthenticationOauth2(issuerUrl: Uri, audience: string, privateKey:
             return temp
         }
 
+    let base64DeserializeCreds data =
+        let bytes = Convert.FromBase64String(data)
+        let decodedData = System.Text.Encoding.UTF8.GetString(bytes)
+        JsonSerializer.Deserialize<Credentials>(decodedData)
+
+    let decodeAndDeserializeCreds (privateKey: Uri) =
+        let split = privateKey.LocalPath.Split(',', 2)
+        let contentType = split[0]
+        let data = split[1]
+        match contentType.ToLowerInvariant() with
+        | "application/json;base64" ->
+            base64DeserializeCreds(data)
+        | _ ->
+            raise <| NotSupportedException($"Content type '{contentType}' is not supported.")
+
+    let deserializeCreds (privateKey: Uri) =
+        backgroundTask {
```

Comment:
I would write it as 
```fsharp
match privateKey.Scheme with
| "file" ->
    openAndDeserializeCreds privateKey.LocalPath
| "data" ->
    decodeAndDeserializeCreds privateKey |> Task.FromResult
| _ ->
    raise <| NotSupportedException $"Scheme '{privateKey.Scheme}' is not supported."
```
and would also skip `.ToLowerInvariant()`

Reply:
I'm having a lot of trouble with this one. The mixing of sync and async is resulting in casting issues.